web/list/rrd/ajax.php

```diff
@@ -9,28 +9,42 @@
 	exit();
 }
 
-if (empty($_POST["period"])) {
-	$period = "daily";
-} else {
-	if (in_array($_POST["period"], ["day", "week", "month", "year"])) {
-		$period = $_POST["period"];
-	} else {
-		$period = "daily";
-	}
-}
+$requestPayload = json_decode(file_get_contents("php://input"), true);
 
-if (empty($_POST["service"])) {
-	$service = "la";
-} else {
-	$service = $_POST["service"];
-}
+$allowedPeriods = ["daily", "weekly", "monthly", "yearly"];
+
+$period =
+	!empty($requestPayload["period"]) && in_array($requestPayload["period"], $allowedPeriods)
+		? $requestPayload["period"]
+		: "daily";
+
+$service = !empty($requestPayload["service"]) ? $requestPayload["service"] : "la";
 
 // Data
 exec(
 	HESTIA_CMD . "v-export-rrd " . quoteshellarg($service) . " " . quoteshellarg($period),
 	$output,
 	$return_var,
 );
+
+if ($return_var != 0) {
+	http_response_code(500);
+	exit("Error fetching RRD data");
+}
+
+$serviceUnits = [
+	"la" => "Points",
+	"mem" => "Mbytes",
+	"net_enp0s1" => "KBytes",
```

Comment:
Changes on the network port the user as it makes more sense to match it against net_ instead define each possible network port name...

Reply:
or net will probally also work

Reply:
Fixed I think